From aa556d82d0ab27a552fbf2e884e8ba93cbe6c9dd Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 5 Dec 2024 15:28:38 +0000
Subject: [PATCH] build: Define MLOBJECT (OCaml object file) as either cmo or
 cmx

This macro expands to either cmo or cmx depending on whether we are
compiling for bytecode or native code.

This is a natural extension of the existing MLARCHIVE macro.

(cherry picked from commit 7b43b226eb3af18a5bd58b2664fb064832404a47)
(cherry picked from commit e6f1a170002bff5db70aa8a8b6987ef1122d61e1)
---
 subdir-rules.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/subdir-rules.mk b/subdir-rules.mk
index e969f357..05221b95 100644
--- a/subdir-rules.mk
+++ b/subdir-rules.mk
@@ -47,10 +47,12 @@ $(top_builddir)/generator/generator:
 
 if !HAVE_OCAMLOPT
 MLARCHIVE = cma
+MLOBJECT = cmo
 LINK_CUSTOM_OCAMLC_ONLY = -output-complete-exe
 BEST = c
 else
 MLARCHIVE = cmxa
+MLOBJECT = cmx
 BEST = opt
 endif
 
